### PR TITLE
fix: disallow control-effects in spawn and new expressions

### LINKED ISSUE #8853: Enforce that spawn and new object expressions cannot have control-effects

### PR DESCRIPTION
**Pull request by magnus-madsen**
Related to #8853 but does not fully solve it

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze This is neither sufficient nor complete. But its better than nothing.

File: main/src/library/Adaptor.flix [hunk 1]
```diff
@@ -40,11 +40,11 @@ mod Adaptor {
     /// Note: Currently requires a Proxy (until Java types support generics).
     ///
     pub def comparator(_: Proxy[a]): Comparator \ IO with Order[a] =
-        let f1 = (o1, o2) -> {
+        let f1 = (o1, o2) -> unchecked_cast ({
```
Comment (mlutze) [inline main/src/library/Adaptor.flix:43]: Why is this one necessary?

File: main/src/ca/uwaterloo/flix/language/phase/Safety.scala [hunk 2]
```diff
@@ -1073,4 +1083,13 @@ object Safety {
   private def isAbstractMethod(m: java.lang.reflect.Method): Boolean =
     java.lang.reflect.Modifier.isAbstract(m.getModifiers)
 
+  /**
+    * Returns the control-effects inside `tpe`.
+    */
+  private def getControlEffs(tpe: Type): Set[Symbol.EffectSym] =
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Safety.scala:1089]: Should this be in Type? We already have several similar functions there.

**Review by mlutze — APPROVED** [review]

File: main/src/library/Adaptor.flix (hunk 1 above)
Comment (magnus-madsen) [reply main/src/library/Adaptor.flix:43]: Fixed.

File: main/src/ca/uwaterloo/flix/language/phase/Safety.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Safety.scala:1089]: I think this function is stupid and should be reworked. So I would keep it here for now.

**Pull request merged**